public/app/features/dashboard/dashgrid/DashboardGrid.tsx

```diff
@@ -383,6 +385,26 @@ const GrafanaGridItem = React.forwardRef<HTMLDivElement, GrafanaGridItemProps>((
   let width = 100;
   let height = 100;
 
+  const anchoredCount = useRef(0);
+
+  const [contextVal, setContextVal] = useState<LayoutItemContextProps>({
```

Comment:
I would add a quick structural annotation for clarity for those less familiar with this pattern - also feel free to ignore, I'm a comment zealot ->

```suggestion
// Use `useRef` to track the count of anchored tooltips or context menus to avoid triggering re-renders when the count changes. Re-render logic in `useState` below.
  const [contextVal, setContextVal] = useState<LayoutItemContextProps>({
```